Sources/ArcGISToolkit/Components/Form/FormInputs/TextInput.swift

```diff
@@ -92,67 +114,99 @@ struct MultiLineTextInput: View {
+                    )
+                }
+            }
+            .accessibilityIdentifier("\(element.label) Text Input")
+            .background(.clear)
+            .focused($isFocused)
+            .keyboardType(keyboardType)
+            .toolbar {
+                ToolbarItemGroup(placement: .keyboard) {
+                    if UIDevice.current.userInterfaceIdiom == .phone, isFocused, fieldType.isNumeric {
+                        positiveNegativeButton
+                        Spacer()
+                    }
                 }
             }
             .scrollContentBackgroundHidden()
             if isFocused && !text.isEmpty && inputModel.isEditable {
                 ClearButton { text.removeAll() }
+                    .accessibilityIdentifier("\(element.label) Clear Button")
             }
         }
-        .background(.clear)
-        .focused($isFocused)
-        .foregroundColor(isPlaceholder ? .secondary : .primary)
-        .onChange(of: model.focusedElement) { focusedElement in
-            // Another form input took focus
-            if focusedElement != element {
-                isFocused  = false
-            }
-        }
-        .onChange(of: model.focusedElement) { focusedElement in
-            // Another form input took focus
-            if focusedElement != element {
-                isFocused  = false
+        .formInputStyle()
+    }
+    
+    /// A Boolean value indicating whether the input is multiline or not.
+    var isMultiline: Bool {
+        element.input is TextAreaFormInput
+    }
+    
+    /// The keyboard type to use depending on where the input is numeric and decimal.
+    var keyboardType: UIKeyboardType {
+        fieldType.isNumeric ? (fieldType.isFloatingPoint ? .decimalPad : .numberPad) : .default
+    }
+    
+    /// The button that allows a user to switch the numeric value between positive and negative.
+    var positiveNegativeButton: some View {
+        Button {
+            if let value = Int(text) {
+                text = String(value * -1)
+            } else if let value = Float(text) {
+                text = String(value * -1)
+            } else if let value = Double(text) {
+                text = String(value * -1)
             }
+        } label: {
+            Image(systemName: "plus.forwardslash.minus")
         }
-        .onChange(of: isFocused) { isFocused in
-            if isFocused && isPlaceholder {
-                isPlaceholder = false
-                text = ""
-            } else if !isFocused && text.isEmpty {
-                isPlaceholder = true
-                text = element.hint
-            }
-            if isFocused {
-                model.focusedElement = element
-            } else if model.focusedElement == element {
-                model.focusedElement = nil
-            }
+    }
+    
+    /// The range of valid values for a numeric input field.
+    var rangeDomain: RangeDomain? {
```

Comment:
You should be able to check the `element.domain` now; I think Ivan fixed it.

Reply:
Yup that seems to work good now.